packages/asset-swapper/src/utils/market_operation_utils/sampler_operations.ts

```diff
@@ -343,181 +381,199 @@ export const samplerOperations = {
+                                            [takerToken, wethAddress, makerToken],
+                                            takerFillAmounts,
+                                        ),
+                                    );
+                                }
+                                return ops;
+                            case ERC20BridgeSource.Kyber:
+                                return samplerOperations.getKyberSellQuotes(makerToken, takerToken, takerFillAmounts);
+                            case ERC20BridgeSource.Curve:
+                                return getCurveAddressesForPair(takerToken, makerToken).map(curveAddress =>
+                                    samplerOperations.getCurveSellQuotes(
+                                        curveAddress,
+                                        MAINNET_CURVE_CONTRACTS[curveAddress].indexOf(takerToken),
+                                        MAINNET_CURVE_CONTRACTS[curveAddress].indexOf(makerToken),
+                                        takerFillAmounts,
+                                    ),
+                                );
+                            case ERC20BridgeSource.LiquidityProvider:
+                                if (liquidityProviderRegistryAddress === undefined) {
+                                    throw new Error(
+                                        'Cannot sample liquidity from a LiquidityProvider liquidity pool, if a registry is not provided.',
+                                    );
+                                }
+                                return samplerOperations.getLiquidityProviderSellQuotes(
+                                    liquidityProviderRegistryAddress,
+                                    makerToken,
+                                    takerToken,
+                                    takerFillAmounts,
+                                );
+                            case ERC20BridgeSource.MultiBridge:
+                                if (multiBridgeAddress === undefined) {
+                                    throw new Error(
+                                        'Cannot sample liquidity from MultiBridge if an address is not provided.',
+                                    );
+                                }
+                                const intermediateToken = getMultiBridgeIntermediateToken(takerToken, makerToken);
+                                return samplerOperations.getMultiBridgeSellQuotes(
+                                    multiBridgeAddress,
+                                    makerToken,
+                                    intermediateToken,
+                                    takerToken,
+                                    takerFillAmounts,
+                                );
+                            case ERC20BridgeSource.Balancer:
+                                const pools = await getBalancerPoolsForPairAsync(takerToken, makerToken);
```

Comment:
this'll probably throw if TheGraph is down. We should handle this so it doesn't take down the entire sampling operation.